src/cljs/main/broadfcui/tooltip.cljs

```diff
@@ -0,0 +1,113 @@
+(ns broadfcui.tooltip
+  (:require
+   [broadfcui.config :as config]
+   [broadfcui.utils :as u]
+   [dmohs.react :as r]
+   ))
+
+(defonce instance nil)
+
+(defn show [id trigger content]
+  (instance :show id trigger content))
+
+(defn remove-tooltip [id]
+  (instance :deref id :trigger))
+
+(r/defc ToolTip
+  {:component-will-mount
+   (fn [{:keys [locals]}]
+     (swap! locals assoc :id (gensym "tooltip-")))
+   :render
+   (fn [{:keys [this props state]}]
+     [:span {:onMouseEnter #(swap! state assoc :show? true)
+             :onMouseLeave #(swap! state dissoc :show?)}
+      (:trigger-element props)])
+   :component-did-update
+   (fn [{:keys [this props state locals]}]
+     (if (:show? @state)
+       (show (:id @locals) (r/find-dom-node this) (:hover-element props))
+       (remove-tooltip (:id @locals))))})
+
+(defn- find-tooltip [tooltips id]
+  (first (filter #(= id (:id %)) tooltips)))
+
+(defn- update-tooltip [tooltips id f & args]
+  (let [tip (find-tooltip tooltips id)
+        new? (if tip false true)
+        tip (if new? {:id id} tip)
+        tooltips (if new? (conj tooltips tip) tooltips)
+        updated (apply f tip args)]
+    (if updated
+      (mapv #(if (= id (:id %)) updated %) tooltips)
+      (filterv #(not= id (:id %)) tooltips))))
+
+(defn- remove-derefed [tooltips]
+  (filterv #(not (empty? (:references %))) tooltips))
+
+(r/defc Container
+  {:show
+   (fn [{:keys [state]} id trigger content]
+     (swap! state update :tooltips update-tooltip id
+            assoc :references #{:trigger} :trigger trigger :content content))
+   :addref
+   (fn [{:keys [state]} id reference]
+     (swap! state update :tooltips update-tooltip id update :references conj reference))
+   :deref
+   (fn [{:keys [state]} id reference]
+     (swap! state update :tooltips
+            (fn [tooltips]
+              (-> tooltips
+                  (update-tooltip id update :references disj reference)
+                  remove-derefed))))
+   :get-initial-state
+   (fn []
+     {:tooltips []})
```

Comment:
Maintaining order ensures that a tooltip displayed later will be on top. The argument could be made that it doesn't matter, but I think this pattern is more generally useful. For example, order probably matters for modals and the same pattern could be used there.